### PR TITLE
Cli restructure

### PR DESCRIPTION
**Pull request by VirajSalaka**

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 1]
```diff
@@ -32,9 +32,11 @@
 import org.wso2.apimgt.gateway.cli.constants.GatewayCliConstants;
 import org.wso2.apimgt.gateway.cli.exception.*;
 import org.wso2.apimgt.gateway.cli.hashing.HashUtils;
+import org.wso2.apimgt.gateway.cli.model.config.BasicAuth;
 import org.wso2.apimgt.gateway.cli.model.config.Config;
 import org.wso2.apimgt.gateway.cli.model.config.ContainerConfig;
 import org.wso2.apimgt.gateway.cli.utils.GatewayCmdUtils;
+import org.wso2.apimgt.gateway.cli.utils.JsonProcessingUtils;
```
Comment (praminda) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java]: Is this import used?

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 2]
```diff
@@ -32,9 +32,11 @@
 import org.wso2.apimgt.gateway.cli.constants.GatewayCliConstants;
 import org.wso2.apimgt.gateway.cli.exception.*;
 import org.wso2.apimgt.gateway.cli.hashing.HashUtils;
+import org.wso2.apimgt.gateway.cli.model.config.BasicAuth;
```
Comment (praminda) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java]: Is this import used? I don't see a reference

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/APIMetadataDTO.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.wso2.apimgt.gateway.cli.model.rest;
```
Comment (praminda) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/APIMetadataDTO.java]: Missing license header

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java [hunk 4]
```diff
@@ -316,30 +315,55 @@ public static void createProjectStructure(String projectName) {
...
+    }
+
+    private static void saveSwaggerDefinitionForSingleAPI(String projectName, ExtendedAPI api){
+        String swaggerString = SwaggerUtils.generateSwaggerString(api);
+        String apiId = HashUtils.generateAPIId( api.getName(), api.getVersion());
+        GatewayCmdUtils.saveSwaggerDefinition(projectName, apiId, swaggerString);
+    }
+
+    /**
+     * Save swagger definition for multiple APIs
```
Comment (praminda) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java]: Need full stop and empty new line after the description 
```suggestion
     * Save swagger definition for multiple APIs.
     
```

**Comment by praminda** [comment]
@VirajSalaka Please remove the unfilled content from the PR template

**Comment by praminda** [comment]
@VirajSalaka Also include the intention of the PR in PR title and the purpose

**Comment by VirajSalaka** [comment]
> @VirajSalaka Please remove the unfilled content from the PR template

**Pull request merged**